aggregator_core/src/datastore.rs

```diff
@@ -3852,6 +3916,10 @@ pub enum Error {
     /// An error occurred while manipulating timestamps or durations.
     #[error("{0}")]
     TimeOverflow(&'static str),
+    /// The version of the DB schema found in `_sqlx_migrations` is not supported by this version of
+    /// Janus.
+    #[error("unsupported schema version {version} / {description}")]
+    UnsupportedSchemaVersion { version: i64, description: String },
```

Comment:
Does this need to be a separately-discriminated error? It's hard to imagine we'll ever do anything with this error other than report it to a human being.

If we don't think we'll make program-logic decisions based on this error discriminant, let's introduce an `Other` value that wraps an `anyhow::Error`.

Reply:
In that case we may as well use `Error::DbState` and represent the error as a `String`.

Reply:
This works, but my point is: I'm not sure we need a separate `DbState` error discriminant, either -- we generate a number of `DbState` errors, but nothing ever consumes a `DbState` error to make a program-logic decision, so `DbState` should ideally also be `Other`.

I don't think we need to clean up existing error discriminants, but there's no reason to introduce additional unnecessary discriminants, either.

[I'm not married to the idea of using `anyhow::Error` as the internal representation--`String` would work too. I guess I see the advantage of `anyhow::Error` as having a nice API & providing error-wrapping.]